quantum/action.c

```diff
@@ -373,7 +373,8 @@ void process_action(keyrecord_t *record, action_t action) {
     if (is_oneshot_layer_active() && event.pressed &&
         (action.kind.id == ACT_USAGE || !(IS_MODIFIER_KEYCODE(action.key.code)
 #    ifndef NO_ACTION_TAPPING
-                                          || (tap_count == 0 && (action.kind.id == ACT_LMODS_TAP || action.kind.id == ACT_RMODS_TAP))
+                                          || ((action.kind.id == ACT_LMODS_TAP || action.kind.id == ACT_RMODS_TAP) &&
+                                                (action.layer_tap.code <= MODS_TAP_TOGGLE || tap_count == 0))
```

Comment:
`tap_count == 0` accounted for mod-tap, but not for OSM. `action.layer_tap.code <= MODS_TAP_TOGGLE` accounts for `MODS_ONESHOT` and `MODS_TAP_TOGGLE`. TBH I don't know how the latter can occur, can't figure a keycode leading to it. But considering it's for mods also I think it makes sense to consider them too.